### ISSUE TITLE
Write a guide demonstrating basic project setup with cabal

### ISSUE THREAD

**Issue by thestaphmonster** [tomjaguarpaw/tilapia#81]
As an extension to [Write a guide for each OS that gets you a basic setup with VS Code and the language server](https://github.com/tomjaguarpaw/tilapia/issues/59).

### Motivation
I recall it being a pain point to have to figure out how to set up a project in a cabal file, in terms of things like:
- Module names/folder structure
- Exposing modules
- How to have one or many executables that depend on a library

Obviously this isn't a massive problem, and it likely didn't take as long to figure out as I'm remembering, but it wouldn't take long to write a small guide. If we can save a few thousand people 30 minutes each, that's worthwhile.

### After reading the guide, a user will be able to...
- Create a compiling Haskell project with two executables which depend on a library
- Understand how to add new, exposed modules to the library
- Understand how to add new, internal modules to the library
- Understand how to add new folders containing more exposed and internal modules to the library
- Understand how to add dependencies to an external package

**Comment by thestaphmonster** [comment]
Let me know if you think this is a good idea, if there's anything I've missed or if it already exists somewhere and I've just not come across it!

**Comment by tomjaguarpaw** [comment]
I think this is a good idea. On the other hand I suspect there may be many such guides floating around on blogs and the like. At some point we should do a "literature review" of what is out there and try to consolidate everything in one place.

**Comment by thestaphmonster** [comment]
Good point. I'll double check what's already there and make sure I'm not repeating existing docs before I start writing anything.

**Comment by tomjaguarpaw** [comment]
If you find existing docs that do some or all of the job then please submit a PR which collects them in a list for easy reference later!

**Comment by thestaphmonster** [comment]
I've not had a look yet, but I did just see [this thread on reddit](https://www.reddit.com/r/haskell/comments/m8q026/first_glances_frustrated_with_introductory/)

> there's no focus on how to even start creating a haskell project, how to structure a project, compiling a project, how to work with the environment, or even how main works. working in ghci seems great, but it's not the same as learning haskell. definitely understand that set theory is important, but that's not introductory material for a language either.
> 
> would be awesome to see an actual tutorial which focused on creating a project, compiling projects, working with the environment, how main works, ideas on structuring a project
